validator/src/main/resources/expected-data-template/python/eks/aws-sdk-call-log.mustache

```diff
@@ -0,0 +1,25 @@
+[{
+  "HostedIn.EKS.Cluster": "^{{platformInfo}}$",
+  "HostedIn.K8s.Namespace": "^{{appNamespace}}$",
+  "K8s.Node": "^i-[A-Za-z0-9]{17}$",
+  "K8s.Pod": "^python-(r-)?app-deployment(-[A-Za-z0-9]*)*$",
+  "K8s.Workload": "^python-app-deployment(-[A-Za-z0-9]*)*$",
+  "Operation": "GET aws-sdk-call",
+  "OTelLib": "^AwsSpanMetricsProcessor$",
+  "Version": "^1$",
+  "aws.span.kind": "^LOCAL_ROOT$"
+},
+{
+  "HostedIn.EKS.Cluster": "^{{platformInfo}}$",
+  "HostedIn.K8s.Namespace": "^{{appNamespace}}$",
+  "K8s.Node": "^i-[A-Za-z0-9]{17}$",
+  "K8s.Pod": "^python-app-deployment(-[A-Za-z0-9]*)*$",
+  "K8s.Workload": "^python-app-deployment(-[A-Za-z0-9]*)*$",
+  "Operation": "GET aws-sdk-call",
+  "OTelLib": "^AwsSpanMetricsProcessor$",
+  "Version": "^1$",
+  "RemoteService": "AWS.SDK.S3",
+  "RemoteOperation": "GetBucketLocation",
+  "RemoteTarget": "e2e-test-bucket-name",
```

Comment:
Yeah, these discrepencies are weird for `::s3:::`.
You’ll notice that in the existing validation logic in `Java`, there is no pattern either

Metrics:
`eks/aws-sdk-call-metric.mustache` -> [e2e-test-bucket-name
](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/eks/aws-sdk-call-metric.mustache#L115-L116)`ec2/aws-sdk-call-metric.mustache` -> [::s3:::e2e-test-bucket-name](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/ec2/aws-sdk-call-metric.mustache#L97-L98)
Logs
`eks/aws-sdk-call-log.mustache` -> [e2e-test-bucket-name](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/eks/aws-sdk-call-log.mustache#L23)
`ec2/aws-sdk-call-log.mustache` -> [e2e-test-bucket-name](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/ec2/aws-sdk-call-log.mustache#L15)
Traces
`eks/aws-sdk-call-trace.mustache` -> [e2e-test-bucket-name](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/eks/aws-sdk-call-trace.mustache#L47C35-L47C55)
`ec2/aws-sdk-call-trace.mustache` -> [^::s3:::e2e-test-bucket-name$](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/ec2/aws-sdk-call-trace.mustache#L45C35-L45C64)

Reply:
Test using latest Staging ADOT Python code
https://github.com/aws-observability/aws-otel-python-instrumentation/actions/runs/8194027328

Reply:
> Yeah, these discrepencies are weird for `::s3:::`. You’ll notice that in the existing validation logic in `Java`, there is no pattern either
> 
> Metrics: `eks/aws-sdk-call-metric.mustache` -> [e2e-test-bucket-name ](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/eks/aws-sdk-call-metric.mustache#L115-L116)`ec2/aws-sdk-call-metric.mustache` -> [::s3:::e2e-test-bucket-name](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/ec2/aws-sdk-call-metric.mustache#L97-L98) Logs `eks/aws-sdk-call-log.mustache` -> [e2e-test-bucket-name](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/eks/aws-sdk-call-log.mustache#L23) `ec2/aws-sdk-call-log.mustache` -> [e2e-test-bucket-name](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/ec2/aws-sdk-call-log.mustache#L15) Traces `eks/aws-sdk-call-trace.mustache` -> [e2e-test-bucket-name](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/eks/aws-sdk-call-trace.mustache#L47C35-L47C55) `ec2/aws-sdk-call-trace.mustache` -> [^::s3:::e2e-test-bucket-name$](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/ec2/aws-sdk-call-trace.mustache#L45C35-L45C64)

Ah I know why it is that way in java, because the add-on hasn't been released yet with that change. 
The PR is out to make the change as soon as the add-on releases. But for python we should use the updated value anyway.

Reply:
Addressed. Changed to use `::s3:::e2e-test-bucket-name`

Reply:
Discussed offline with @jj22ee and he clarified the semantics of the validation templates. 
In the template `"Foo": "bar"` is a "contains" check whereas `"Foo": "^bar$"` means an "exact match". This explains why `"RemoteTarget: "e2e-test-bucket-name"` check was passing when the actual value was in fact `::s3:::e2e-test-bucket-name`.